src/trajectory.jl

```diff
@@ -178,11 +193,11 @@ function transition(
 
     local α, nα
     while s == 1 && j <= nt.max_depth
-        v = rand(rng, [-1, 1])
+        v = rand(rng, -1:2:1)
```

Comment:
```julia
julia> @benchmark rand(-1:2:1)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     4.030 ns (0.00% GC)
  median time:      4.608 ns (0.00% GC)
  mean time:        4.636 ns (0.00% GC)
  maximum time:     14.526 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000

julia> @benchmark rand((-1,1))
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     2.558 ns (0.00% GC)
  median time:      3.120 ns (0.00% GC)
  mean time:        3.137 ns (0.00% GC)
  maximum time:     13.086 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000
```
https://github.com/TuringLang/AdvancedHMC.jl/blob/mt/perf_boost/src/trajectory.jl#L196

Obviously this is a completely negligible difference in performance, but I do think it's slightly easier to read.

There is also
```julia
julia> fastrand(rng = Random.GLOBAL_RNG) = (rand(rng, Int32) & 1) << 1 - 1
fastrand (generic function with 2 methods)

julia> fastrand()
-1

julia> fastrand()
1

julia> @benchmark fastrand()
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     2.542 ns (0.00% GC)
  median time:      3.108 ns (0.00% GC)
  mean time:        3.126 ns (0.00% GC)
  maximum time:     12.777 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000
```
Seems about the same fast as the tuple for MersenneTwister, but does a little better with some other RNGs:
```julia
julia> using RandomNumbers

julia> rng = RandomNumbers.PCG.PCGStateUnique(PCG.PCG_RXS_M_XS)
RandomNumbers.PCG.PCGStateUnique{UInt64,Val{:RXS_M_XS},UInt64}(0xc15dfaa9205f04f6)

julia> @benchmark rand($rng, -1:2:1)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     10.370 ns (0.00% GC)
  median time:      11.244 ns (0.00% GC)
  mean time:        11.217 ns (0.00% GC)
  maximum time:     21.496 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     999

julia> @benchmark rand($rng, (-1,1))
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     2.268 ns (0.00% GC)
  median time:      2.290 ns (0.00% GC)
  mean time:        2.336 ns (0.00% GC)
  maximum time:     11.914 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000

julia> @benchmark fastrand($rng)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     1.923 ns (0.00% GC)
  median time:      1.968 ns (0.00% GC)
  mean time:        1.989 ns (0.00% GC)
  maximum time:     11.560 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000
```
So I'd suggest either going with the tuple or the bit hack.